### PR TITLE
Document methods for creating data pipeline interface objects

### PR DESCRIPTION
**Pull request by shsms**
Also clarify in `client` package that it is a low-level one.

### CODE REVIEW

File: src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py [hunk 1]
```diff
@@ -53,7 +53,26 @@ class EVChargerData:
 
 
 class EVChargerPool:
-    """Interactions with EV Chargers."""
+    """An interface for interaction with pools of EV Chargers.
+
+    !!! note
+        `EVChargerPool` instances are not meant to be created directly by users. Use the
+        [`microgrid.ev_charger_pool`][frequenz.sdk.microgrid.ev_charger_pool] method for
+        creating an instance.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py]: Optionally to be consistent:
```suggestion
        creating `EVChargerPool` instances.
```

File: src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py [hunk 2]
```diff
@@ -64,6 +83,11 @@ def __init__(
     ) -> None:
         """Create an `EVChargerPool` instance.
 
+        !!! note
+            `EVChargerPool` instances are not meant to be created directly by users. Use
+            the [`microgrid.ev_charger_pool`][frequenz.sdk.microgrid.ev_charger_pool]
+            method for creating an instance.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py]:
```suggestion
            method for creating `EVChargerPool` instances.
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
A couple of optional cosmetics comments to check for and LGTM

**Comment by shsms** [comment]
Thanks Dani, both are done.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**